Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -60,14 +60,14 @@ struct ClangModuleBuildMetadata {
     var inputs: [String] {
         return module.recursiveDependencies.flatMap { module in
             switch module {
-            case let module as ClangModule:
-                 let product = Product(name: module.name, type: .Library(.Dynamic), modules: [module])
-                return product.targetName
-            case let module as CModule:
-                return module.targetName
-            case let module as SwiftModule:
-                return module.targetName
-            default:
+              case let module as ClangModule:
+                let product = Product(name: module.name, type: .Library(.Dynamic), modules: [module])
+                return prefix.appending(product.outname).asString
+              case _ as CModule:
```

Comment:
I _think_ there is a `case is <Type>` form? If so, probably worth a bug on swift.org that the compiler should just suggest that form instead.